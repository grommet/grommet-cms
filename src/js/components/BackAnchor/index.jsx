/* @flow */
import React from 'react';
import Box from 'grommet/components/Box';
import Anchor from 'grommet/components/Anchor';
import LinkPreviousIcon from 'grommet/components/icons/base/LinkPrevious';

export default function BackAnchor(props: {
  title: string,
  onClick: Function
}) {
  return (
    <Box>
      <Anchor
        label={props.title}
        onClick={props.onClick}
        icon={<LinkPreviousIcon />}
      />
    </Box>
  );
}
