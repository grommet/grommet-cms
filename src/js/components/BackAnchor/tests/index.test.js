import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';
import { spy } from 'sinon';
import BackAnchor from '../index';
import props from './__mocks__/backAnchorMocks';

describe('<BackAnchor />', () => {
  it('should render with default props', () => {
    const mockOnClick = spy();
    const wrapper = shallow(
      <BackAnchor
        {...props}
        onClick={mockOnClick}
      />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
