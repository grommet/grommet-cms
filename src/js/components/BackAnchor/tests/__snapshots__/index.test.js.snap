exports[`<BackAnchor /> should render with default props 1`] = `
<Box
  announce={false}
  direction="column"
  focusable={true}
  pad="none"
  responsive={true}
  tag="div">
  <Anchor
    animateIcon={true}
    icon={
      <LinkPrevious
        responsive={true} />
    }
    label="Posts"
    method="push"
    onClick={[Function]}
    tag="a" />
</Box>
`;
